lumispy/utils/io.py

```diff
@@ -0,0 +1,131 @@
+# along with LumiSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+
+SAVETXT_DOCSTRING = \
+    """
+    Writes single spectra to a two-column data file with signal axis as
+        X and data as Y.
+    Writes linescan data to file with signal axis as first row and
+        navigation axis as first column (flipped if `transpose=True`)."""
+
+SAVETXT_PARAMETERS = \
+    """
+    Parameters
+    ----------
+    filename : string
+    fmt : str or sequence of strs, optional
+        A single or sequence of format strings. Default is '%.5f'.
+    delimiter : str, optional
+        String or character separating columns. Default is ','
+    axes : bool, optional
+        If True (default), include axes in saved false. If False, save the data
+        array only.
+    transpose : bool, optional
+        If True, transpose data array and exchange axes. Default is false.
+        Ignored for single spectra.
+    **kwargs 
+        Takes any additional arguments of numpy.loadtxt, e.g. `newline`
+        `header`, `footer`, `comments`, or `encoding`.
+
+    See also
+    --------
+    numpy.savetxt"""
+
+SAVETXT_EXAMPLE = \
+    """
+    Examples
+    --------
+    >>> import lumispy as lum
+    >>> import numpy as np
+    
+    # Spectrum:
+    >>> s = lum.signals.LumiSpectrum(np.arange(5))
+    >>> lum.savetxt(s, 'spectrum.txt')
+    # 0.00000	0.00000
+    # 1.00000	1.00000
+    # 2.00000	2.00000
+    # 3.00000	3.00000
+    # 4.00000	4.00000
+    
+    # Linescan:
+    >>> l = lum.signals.LumiSpectrum(np.arange(25).reshape((5,5)))
+    >>> lum.savetxt(l, 'linescan.txt')
+    # 0.00000	0.00000	1.00000	2.00000	3.00000	4.00000
+    # 0.00000	0.00000	5.00000	10.00000	15.00000	20.00000
+    # 1.00000	1.00000	6.00000	11.00000	16.00000	21.00000
+    # 2.00000	2.00000	7.00000	12.00000	17.00000	22.00000
+    # 3.00000	3.00000	8.00000	13.00000	18.00000	23.00000
+    # 4.00000	4.00000	9.00000	14.00000	19.00000	24.00000
+    """
+
+
+def savetxt(S, filename, fmt='%.5f', delimiter='\t', axes=True,
+            transpose=False, **kwargs):
+    """Writes signal object to simple text file.
+    %s
+    Writes image to file with the navigation axes as first column and first
+        row.
+    Writes 2D data (e.g. map of a fit parameter value) to file with the signal
+        axes as first column and first row.
+    %s
+    %s
+    """
+    nav_axes = S.axes_manager.navigation_axes
+    sig_axes = S.axes_manager.signal_axes
+    dim = len(nav_axes) + len(sig_axes)
+    # Write single spectrum or data with only navigation axis
+    if dim == 1:
+        if axes:
+            if len(sig_axes) == 1:
+                X = sig_axes[0].axis
+            else:
+                X = nav_axes[0].axis
+            output = np.array([X,S.data]).T
+        else:
+            output = S.data
+        np.savetxt(filename, output, fmt=fmt, delimiter=delimiter, **kwargs)
+    # Write linescan or matrix
+    elif dim == 2:
```

Comment:
I am not sure (and I am happy for you to keep this like that, but what about creating 2 sub-functions for each case?
So:
```
if dim ==  1:
    savetxt_1d(*)
elif dim == 2:
     savetxt_2d(*)
```
This would allow us to add other dimensions in the future in a clean and easy way?

Reply:
It is not so much code that I consider it worth splitting, and even a third dimension could be easily added - anything beyond will get hard to handle in a text file.